rclpy/src/rclpy/_rclpy.c

```diff
@@ -1332,15 +1437,14 @@ rclpy_create_service(PyObject * Py_UNUSED(self), PyObject * args)
       PyErr_Format(PyExc_ValueError,
         "Failed to create service due to invalid topic name '%s': %s",
         service_name, rcl_get_error_string_safe());
-      rcl_reset_error();
-      return NULL;
+    } else {
+      PyErr_Format(PyExc_RuntimeError,
+        "Failed to create service: %s", rcl_get_error_string_safe());
     }
-    PyErr_Format(PyExc_RuntimeError,
-      "Failed to create service: %s", rcl_get_error_string_safe());
+    Py_DECREF(service);
```

Comment:
This line seems to be the reason for this new test failure: http://ci.ros2.org/view/nightly/job/nightly_win_deb/lastCompletedBuild/testReport/(root)/projectroot/rclpytests/